pkg/cmd/provisioning/nutanix/create_shared_secrets.go

```diff
@@ -0,0 +1,209 @@
+package nutanix
+
+import (
+	"fmt"
+	credreqv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/cloud-credential-operator/pkg/cmd/provisioning"
+	"github.com/spf13/cobra"
+)
+
+const (
+	manifestsDirName = "manifests"
+
+	secretManifestsTemplate = `apiVersion: v1
+ kind: Secret
+ metadata:
+   name: %s
+   namespace: %s
+ data:
+  NUTANIX_ENDPOINT: %s
+  NUTANIX_PORT: %s
+  NUTANIX_USER: %s
+  NUTANIX_PASSWORD: %s`
```

Comment:
Currently Nutanix PC APIs client only supports using the credentials with username/password. I see some other openshift platforms also have username/password in their cloud-crendential secrets. Can you clarify why it's harder to migrate to future authentication schemes?

Reply:
We can move the ENDPOINT and PORT to the Infrastructure CR. But I see some other openshift platforms also putting their endpoints/port/url, etc. in the cloud-credentials secrets.

Reply:
@joelddiaz I see from the previous slack discussion thread, you mentioned that Openshift only allows all the new platforms to use the “Manual” credentialMode, and the reason is: "the direction is to not introduce new in-cluster credentials handling for new platforms”.  But only having the “Manual” credentialMode is too tedious and error-prone for the users when creating and updating clusters, even with the "ccoctl". The IPI is supposed to make the cluster creation/updating automatic, and relieve users burden to do any manual work. Also when the cloud-credentials are changed for a cluster, the cluster admin will have to manually change all the cloud-credentials secrets in the cluster. That is tedious and error-prone for the users, as well. I think the CCO (cloud-credential-operator)’s major functionality is to automatically manage/sync the cloud-credentials via the CredentialsRequest CRs. I wonder why the RH team does not allow new platforms to have the Passthrough/Mint modes? Do Openshift wants to phase out the CCO?

Reply:
Yes, managing the credentials from outside the cluster is more tedious. This was a decision made by the OpenShift architects (for sound security-conscious reasons). I'm just implementing the direction they handed down.

The goal is to phase out in-cluster handling and management of cloud/platform credentials.

Reply:
> We can move the ENDPOINT and PORT to the Infrastructure CR. But I see some other openshift platforms also putting their endpoints/port/url, etc. in the cloud-credentials secrets.

Endpoint and port information are not part of the credentials. It is preferable to not mix configuration with credentials.
When supporting alternative API endpoints for AWS (and Azure for that matter), we don't put this URL information in the Secret, we store it in the Infrastructure resource so that all operators have a common place to figure out where they should be making API calls into.

Reply:
OK, we will move the endpoints/port/url, etc. info to Infrastructure CR providerSpec.

Reply:
Thanks for clarification of why Openshift requiring new platforms only have the "manual" credentials mode.

Reply:
Regarding the nutanix-credentials secret data format, I had a look at how AlibabaCloud is doing and think we can do something similar to make the nutanix-credentials data format configurable and extensible.

Reply:
@joelddiaz here's a PR for adding endpoint and port to PlatformStatus in API https://github.com/openshift/api/pull/1116

Reply:
@joelddiaz Thanks for the perspective. We intend to make those changes in a future PR. Until then can we not block the review?

Reply:
@thunderboltsid Joel is no longer on the team. I will take it from here. Just curious, is there anything blocking you from including secret format changes in this PR? 

Also, it will be great if you can add some docs. You can refer to Alibaba cloud and other ccoctl docs [here](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md#alibaba-cloud).

Reply:
@akhil-rane that also requires changes to API, Changes to the operators that consume the Secret, etc. Let's take this iteratively.

Reply:
I believe changes to the API will be in nutanix? Yes, that can be handled in future PRs. 

Is it possible to move endpoint and port to infrastructure resource and remove them from credentials secrets? Or do you want to handle it later?

Reply:
@akhil-rane those changes will be handled in future PRs.